src/v/raft/consensus.cc

```diff
@@ -529,20 +529,41 @@ ss::future<result<model::offset>> consensus::linearizable_barrier() {
     co_return ret_t(snapshot.linearizable_offset);
 }
 
+ss::future<result<replicate_result>> chain_stages(replicate_stages stages) {
+    return stages.request_enqueued.then_wrapped(
+      [f = std::move(stages.replicate_finished)](
+        ss::future<> enqueued) mutable {
+          if (enqueued.failed()) {
+              return f.discard_result()
+                .handle_exception([](const std::exception_ptr& e) {
+                    vlog(
+                      raftlog.debug,
+                      "ignorring replicate second stage exception - {}",
+                      e);
+                })
+                .then([] {
+                    return result<replicate_result>(
+                      make_error_code(errc::replicate_first_stage_exception));
+                })
+                .finally([enqueued = std::move(enqueued)]() mutable {
+                    return std::move(enqueued);
+                });
```

Comment:
I'm not sure what the return value intention is here. I think that replicate_first_stage_exception is always thrown away and the enqueued exceptional future will always be returned.

Reply:
you are right, I misread the docs, it will ignore returned value but not the exceptional future.